src/Engine/Audio/SteamAudioInterface.cpp

```diff
@@ -1,46 +1,107 @@
 #include "SteamAudioInterface.hpp"
-
-#ifdef USINGMEMTRACK
-#include <MemTrackInclude.hpp>
-#endif
+#include <assert.h>
 
 SteamAudioInterface::SteamAudioInterface() {
-    context = nullptr;
-    playerDir = new IPLVector3{ 0, 0, 0 };
-    playerPos = new IPLVector3{ 0, 0, 0 };
-    playerUp = new IPLVector3{ 0, 0, 0 };
+    simSettings.sceneType = IPL_SCENETYPE_PHONON;
+    simSettings.numRays = 12400;
+    simSettings.numDiffuseSamples = 32;
+    simSettings.numBounces = 1;
+    simSettings.irDuration = 1.0;
+    simSettings.ambisonicsOrder = 0;
+    simSettings.maxConvolutionSources = 5;
+    context = IPLContext{ nullptr, nullptr, nullptr };
 }
 
 SteamAudioInterface::~SteamAudioInterface() {
-    delete context;
-    // Might have to remove these depending on how the pointers are sent to this class
-    delete playerDir;
-    delete playerPos;
-    delete playerUp;
+
+}
+
+void SteamAudioInterface::CreateScene(uint32_t numMaterials) {
+    IPLerror err = iplCreateScene(context, NULL, simSettings, numMaterials, &scene);
+}
+
+void SteamAudioInterface::FinalizeScene(IPLFinalizeSceneProgressCallback progressCallback) {
+    iplFinalizeScene(scene, NULL);
+}
+
+SteamAudioInterface::SaveData SteamAudioInterface::SaveFinalizedScene() {
+    SaveData saveData;
+
+    // Call once first only to figure out the size of the saved scene.
+    saveData.sceneSize = iplSaveFinalizedScene(scene, NULL);
+
+    // Actually save the scene.
+    saveData.scene = new IPLbyte[saveData.sceneSize];
+    iplSaveFinalizedScene(scene, saveData.scene);
+
+    return saveData;
 }
 
-void SteamAudioInterface::Process(float* input, uint32_t samples,  IPLVector3 * sourcePosition, float sourceRadius) {
+void SteamAudioInterface::LoadFinalizedScene(SaveData data) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'LoadFinalizedScene' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)